### PR TITLE
Updating route blueprint to write to router when dummy flag is used

### LINKED ISSUE #4404: Generating routes with `--dummy` does not update `tests/dummy/app/router.js`
Because the dummy folder is contained in an addon, the `router.js` is currently not updated when generating a new route inside a dummy app. We should account for this by checking for the dummy flag [here](https://github.com/ember-cli/ember-cli/blob/master/blueprints/route/index.js#L63) and [here](https://github.com/ember-cli/ember-cli/blob/master/blueprints/route/index.js#L76)

### PR DESCRIPTION
**Pull request by gmurphey**
Fixes #4404. /cc @trabus

### CODE REVIEW

**Comment by gmurphey** [comment]
I also did a bit of refactor to the route blueprint to DRY up the `afterInstall` / `afterUninstall` hooks.

**Comment by gmurphey** [comment]
Not sure what's going on with AppVeyor. I'll try to hop on my Windows machine later tonight to figure this out.

File: tests/acceptance/addon-dummy-generate-test.js [hunk 1]
```diff
@@ -272,6 +275,13 @@ describe('Acceptance: ember generate in-addon-dummy', function() {
       assertFile('tests/dummy/app/templates/foo/bar.hbs', {
         contains: '{{outlet}}'
       });
+      assertFile('tests/dummy/app/router.js', {
+        contains: [
+          "this.route('foo', function() {" + EOL,
```
Comment (trabus) [inline tests/acceptance/addon-dummy-generate-test.js]: I don't think asserting the EOL's is necessary, probably what is tripping up Appveyor.

Comment (trabus) [inline tests/acceptance/addon-dummy-generate-test.js]: Hmm, I guess we are asserting them on other tests, nevermind.

**Comment by stefanpenner** [comment]
```
not ok 6 PhantomJS 2.0 - JSHint - .: failing test
 
    ---
        actual: >
            false
        expected: >
            true
        message: >
            test should fail to confirm ember test exit code
        Log: >
    ...
 
ok 7 PhantomJS 2.0 - JSHint - unit: unit/some-test.js should pass jshint
 
 
```

File: tests/acceptance/addon-dummy-generate-test.js (hunk 1 above)
Comment (gmurphey) [inline tests/acceptance/addon-dummy-generate-test.js]: I'm wondering if using `EOL` in tests works with code generated by ember-cli/ember-router-generator. I'm not seeing `EOL` included in anything that's created with `recast`.

**Comment by gmurphey** [comment]
This should be ready for review now. I've removed `EOL` from the `router.js` tests, and that seems to fix everything. I looked and couldn't find another instance of using `EOL` in tests against `router.js`, so I'm assuming ember-router-generator only uses Unix style line-endings.

File: blueprints/route/index.js [hunk 2]
```diff
@@ -61,49 +61,53 @@ module.exports = {
...
+  var entity = options.entity;
 
-      this.ui.writeLine('updating router');
-      this._writeStatusToUI(chalk.red, 'remove route', entity.name);
-    }
+  if (this.shouldTouchRouter(entity.name, options)) {
+    writeRoute(action, entity.name, options);
+
+    this.ui.writeLine('updating router');
+    this._writeStatusToUI(chalk.red, action + ' route', entity.name);
```
Comment (trabus) [inline blueprints/route/index.js]: Is there a specific reason we're using red for adding and removing routes? Might be confusing for users.

**Comment by trabus** [comment]
Aside from the status color issue, the rest looks :+1: to me.

File: blueprints/route/index.js (hunk 2 above)
Comment (gmurphey) [inline blueprints/route/index.js]: Good point -- I'll update this now!

**Pull request merged**